src/app/task-list/task-list.component.ts

```diff
@@ -23,26 +25,27 @@ export class TaskListComponent {
     ])
   });
 
-  constructor(private taskDataService: TaskDataService, private fb: FormBuilder) { }
-
+  constructor(private taskDataService: TaskDataService, private fb: FormBuilder, private userDataService: UserDataService) { }
 
-  public addTask(task: string): void {
+  public addTask(taskValue: string): void {
+    let task = new TaskItem(this.tasks.length, taskValue);
     this.taskDataService.addTask(task);
     this.tasks.push(task);
+    this.taskForm.reset();
   }
 
-  public deleteTask(task: string):void
-  {
-    this.taskDataService.deleteTask(task);
+  public deleteTask(taskIndex: number): void {
+    this.taskDataService.deleteTask(taskIndex);
     for (let i = 0; i < this.tasks.length; i++) {
-      if (this.tasks[i] == task) {
+      if (this.tasks[i].id === taskIndex) {
```

Comment:
We can find index as we did in editTask below

Reply:
Fix is done.